File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net;
using System.Numerics;
using System.Text;

namespace ConsoleApp1
{
    class Program
    {
        static int n;
        static int m;
        static int v;
        static bool[] dfsCheck = new bool[1001];
        static bool[] bfsCheck = new bool[1001];
        static bool[,] tree = new bool[1001, 1001];

        static void Main(string[] args)
        {
            string[] nmv = Console.ReadLine().Split();
            n = int.Parse(nmv[0]);
            m = int.Parse(nmv[1]);
            v = int.Parse(nmv[2]);

            for(int i = 0; i < m; i++)
            {
                string[] input = Console.ReadLine().Split();

                tree[int.Parse(input[0]), int.Parse(input[1])] = true;
                tree[int.Parse(input[1]), int.Parse(input[0])] = true;
            }

            DFS(v);
            Console.WriteLine();
            BFS();
        }

        static void DFS(int vv)
        {
            Stack<int> st = new Stack<int>();

            Console.Write($"{vv} ");
            st.Push(vv);
            dfsCheck[vv] = true;

            for(int i = 1; i <= n; i++)
            {
                if (dfsCheck[i] || !tree[vv, i])
                    continue;

                DFS(i);
            }
        }

        static void BFS()
        {
            Queue<int> q = new Queue<int>();
            q.Enqueue(v);
            bfsCheck[v] = true;
            
            while(q.Count > 0)
            {
                var front = q.Dequeue();

                Console.Write($"{front} ");
                
                for(int i = 1; i <= n; i++)
                {
                    if (bfsCheck[i] || !tree[front, i])
                        continue;

                    q.Enqueue(i);
                    bfsCheck[i] = true;
                }
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <bits/stdc++.h>

using namespace std;

int board[1001][1001];
int check[1001];
int n, m, v;

void Input() {
	cin >> n >> m >> v;

	for (int i = 0; i < m; i++) {
		int x, y;

		cin >> x >> y;

		board[x][y] = 1;
		board[y][x] = 1;
	}

}

void DFS(int temp) {
	stack<int> s;

	s.push(temp);

	check[temp] = 1;

	cout << temp << ' ';
	for (int i = 1; i <= n; i++) {
		if (board[temp][i] == 0 || check[i] == 1) continue;

		check[i] = 1;
		DFS(i);
	}
}

void BFS() {
	memset(check, 0, sizeof(check));
	queue<int> q;

	q.push(v);

	check[v] = 1;

	while (!q.empty()) {
		int cur = q.front();
		q.pop();

		cout << cur << ' ';

		for (int i = 1; i <= n; i++) {
			if (board[cur][i] == 0 || check[i] == 1) continue;

			check[i] = 1;
			q.push(i);
		}
	}
}



int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	Input();
	DFS(v);
	cout << endl;
	BFS();

}